src/main/java/seedu/address/logic/parser/LogCommandParser.java

```diff
@@ -49,16 +49,17 @@ public LogCommand parse(String args) throws ParseException {
         try {
             nric = new Nric(nricString);
         } catch (IllegalArgumentException e) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AppointmentCommand.MESSAGE_USAGE));
+            throw new ParseException("NRIC provided is invalid.");
         }
 
         // Parse the date and time
         String dateTimeString = argParts[1] + " " + argParts[2];
         LocalDateTime dateTime;
         try {
-            dateTime = LocalDateTime.parse(dateTimeString, DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm"));
+            dateTime = LocalDateTime.parse(dateTimeString, DateTimeFormatter.ofPattern("dd-MM-uuuu HH:mm")
+                    .withResolverStyle(ResolverStyle.STRICT));
         } catch (Exception e) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, LogCommand.MESSAGE_USAGE));
+            throw new ParseException("Date and time provided are invalid.");
```

Comment:
More specific error messages is great so that the user understands what is wrong with their input